File: Backend/src/main/java/com/Trabajo_Final_Beltran/dto/response/PedidoDetalleResponse.java
package com.Trabajo_Final_Beltran.dto.response;



import com.Trabajo_Final_Beltran.enums.EstadoPago;
import com.Trabajo_Final_Beltran.enums.EstadoPedido;
import com.Trabajo_Final_Beltran.enums.MetodoPago;
import com.Trabajo_Final_Beltran.enums.TipoEntrega;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class PedidoDetalleResponse {

  private Long id;

  private String numeroPedido;

  private EstadoPedido estado;

  private EstadoPago estadoPago;

  private MetodoPago metodoPago;

  private String urlPago;

  private LocalDateTime fechaHora;

  private BigDecimal total;

  private TipoEntrega tipoEntrega;

  private String nombreCliente;

  private String telefonoCliente;

  private String direccionCliente;

  private List<DetallePedidoResponse> detalles;
  
  private String codigoCuponAplicado; 
  
  private BigDecimal montoDescuento;  
  
  private BigDecimal totalConDescuento;

}

File: frontEnd/src/app/core/models/pedido-detalle-response.ts
import { DetallePedidoResponse } from './detalle-pedido-response';
import { EstadoPago } from './enums/estado-pago.enum';
import { EstadoPedido } from './enums/estado-pedido.enum';
import { MetodoPago } from './enums/metodo-pago.enum';
import { TipoEntrega } from './enums/tipo-entrega.enum';

export interface PedidoDetalleResponse {
    id: number;
    numeroPedido: string;
    estado: EstadoPedido;
    estadoPago: EstadoPago;
    metodoPago: MetodoPago;
    urlPago: string | null;
    fechaHora: string;
    total: number;
    tipoEntrega: TipoEntrega;
    nombreCliente: string;
    telefonoCliente: string;
    direccionCliente: string | null;
    detalles: DetallePedidoResponse[];
    codigoCuponAplicado: string | null;
    montoDescuento: number | null;
    totalConDescuento: number | null;
}
